Src/GoogleApis.Core/Apis/Http/ConfigurableMessageHandler.cs

```diff
@@ -64,22 +74,115 @@ public class ConfigurableMessageHandler : DelegatingHandler
         private readonly IList<IHttpExecuteInterceptor> executeInterceptors =
             new List<IHttpExecuteInterceptor>();
 
-        /// <summary>Gets a list of <see cref="IHttpUnsuccessfulResponseHandler"/>.</summary>
-        public IList<IHttpUnsuccessfulResponseHandler> UnsuccessfulResponseHandlers
+        /// <summary>
+        /// Gets a copied list of <see cref="IHttpUnsuccessfulResponseHandler"/>.
+        /// <remarks>
+        /// Since version 1.10, the return value had been changed from <c>IList</c> to <c>IEnumerable</c> in order to
+        /// keep this class thread-safe. The returned list is also a copied version of the handlers. More info is
+        /// available on <a href="https://github.com/google/google-api-dotnet-client/issues/592">Issue 592</a>.
+        /// </remarks>
+        /// </summary>
+        public IEnumerable<IHttpUnsuccessfulResponseHandler> UnsuccessfulResponseHandlers
+        {
+            get
+            {
+                lock (unsuccessfulResponseHandlersLock)

```

Comment:
1. Regarding SyncRoot, I agree. I will do that. (in general it's not a good practice, but since here that list and its SyncRoot won't be available to anyone, I agree with you).

2.  I'm OK with adding a Get[Copied\Snapshotted]UnsuccessfulResponseHandlers(), but I'm also fine with leaving it like that, since while you get the list you can't really add or remove stuff from it, so it gives you just the option to look in it. Yeah, I do agree that if someone stores that enumerable and later on iterate over it again, he/she will get the same list (since it's unmodified).
I'm leaning toward leave it like that. You?

3. We can't. Keeping the old API doesn't maintain thread-safety. So it's better just deleting it with all the sadness of breaking users.

I'll change the SyncRoot, after we will agree on 2-3.
Thanks Jan for the immediate feedback. Enjoy the wkd :)